test/cli/daemon.js

```diff
@@ -88,18 +87,16 @@ describe('daemon', () => {
     }).catch((err) => done(err))
   })
 
-  // TODO: test fails
-  it.skip('should handle SIGINT gracefully', function (done) {
-    this.timeout(20 * 1000)
+  it('should handle SIGINT gracefully', function (done) {
+    this.timeout(100 * 1000)
 
     testSignal(ipfs, 'SIGINT').then(() => {
       checkLock(repoPath, done)
     }).catch(done)
   })
 
-  // TODO: test fails
-  it.skip('should handle SIGTERM gracefully', function (done) {
-    this.timeout(20 * 1000)
+  it('should handle SIGTERM gracefully', function (done) {
+    this.timeout(100 * 1000)
```

Comment:
Why does this take so much time?

Reply:
It needs to start and init the repo for the IPFS node each time, which seems to be slow.

Reply:
The other tests take 40s or less, why 100s in this one?

Reply:
I wasn't able to get it to work with anything below 60 on my box, 100 seems to be the most stable.

I'm guessing that it is because it has to:
- start the daemon
- init the repo
- send the signal and wait for daemon to clean up and exit

The 40 (in i.e. block tests) https://github.com/ipfs/js-ipfs/blob/ac95601fe88a25299dfb1eb00d4c8688f0a4af0e/test/cli/block.js#L15, are coupled with the 30 https://github.com/ipfs/js-ipfs/blob/104ef1ef6cc64c09ec886f67c28e9b5d37bc9e66/test/utils/on-and-off.js#L20 from the `before` hook that gets the daemon started and initialized.

So I think it currently is more like 70 + ~30 for graceful shutdown. I think we can tweak the 30 a little bit.

Reply:
ok